src/utils.js

```diff
@@ -0,0 +1,105 @@
+import gridConstants from './constants/grid'
+
+/**
+ * 
+ * @param {Array.<Array.<Longitude, Latitude>>} extent Extent is defined by left bottom and right top lon/lat.
+ * @returns {string}
+ */
+export const getExtentID = (extent = gridConstants.LEVEL_BOUNDARIES) => {
+    return `${extent[0][0]}${extent[0][1]}${extent[1][0]}${extent[1][1]}`
+}
+
+/**
+ * https://stackoverflow.com/questions/11832914/round-to-at-most-2-decimal-places-only-if-necessary?page=1&tab=votes#tab-top
+ * @param {Number} num 
+ * @param {Number} scale 
+ * @returns {Number}
+ */
+export const roundNumber = (num, scale) => {
```

Comment:
I give up on trying to divine all the reasoning for this. At least add a todo to get to this later